### ISSUE TITLE
[UI/Gameplay] Villager Dragging and Camera Movement Malfunctions

### ISSUE THREAD

**Issue by AizenGetsu93** [seanmenzies/ShoniIsland#103]
**Issue 1:** The Game Hand and UI Hand fail to function when attempting to drag villagers ( Ravenna or Aldith) to designated locations to work in Slave Mode, such as building a Small Wood Storage.
**Issue 2:** The camera becomes unresponsive to both keyboard (WASD keys) and mouse inputs, halting any ability to adjust the view.

Severity:
Critical – Prevents assigning tasks to villagers, halts progression, and severely impacts gameplay usability.

Expected Results:

- The Game Hand and UI Hand should allow seamless dragging of villagers to designated locations.
- The camera should remain responsive to WASD keys and mouse inputs for smooth navigation.

Actual Results:

- The Game Hand and UI Hand fail to drag villagers, leaving tasks like building a Small Wood Storage incomplete.
- The camera stops responding to WASD keys and mouse movements, effectively freezing navigation.

Steps to Reproduce:

1. Start the game and assign villagers to build a Small Wood Storage or use magic on the totem.
2. Attempt to drag Ravenna or Aldith (Any Villager who is Gifted in such Assignments) to the designated location using the Game Hand or UI Hand (The Game Hand will swap to the UI HAnd when trying to Click and Drag Villager to set work Location).
3. Observe the unresponsiveness of the Game Hand and UI Hand.
4. Try moving the camera using the WASD keys and mouse, and observe the camera becoming unresponsive.

Affected Version:
SHONI ISLAND BETA BUILD

Repro Frequency:
100% – Occurs consistently when attempting to drag villagers to designated locations.

Environment:
MSI Raider GE76
12th Gen Intel(R) Core(TM) i7-12700H 2.70 GHz
NVIDIA GeForce RTX 3080 Laptop GPU
Killer(R) Wi-Fi 6E AX1675i 160MHz Wireless Network Adapter (211NGW)
Windows 11

**Comment by seanmenzies** [comment]
Tried to replicate using your steps but all good on my end. Try to provide more details of what happened before the bug occurs.

**Comment by AizenGetsu93** [comment]
Yes, I do not know why it was doing this. It was right after destroying the ship and placing the totem. I would later designate the villagers to their assigned roles (Gifted Builder or Gifted Magician) and right after, the game was a bit buggy. It did not happen on the latest bug that I created last night. Everything was running was running smoothly but Villagers were traversing within the map.
